bin/runtime/src/lib.rs

```diff
@@ -709,14 +709,27 @@ parameter_types! {
     pub CodeHashLockupDepositPercent: Perbill = Perbill::from_percent(30);
 }
 
+// The filter for the runtime calls that are allowed to be executed by contracts.
+// Currently we allow only staking and nomination pools calls.
+pub struct ContractsCallRuntimeFilter;
+
+impl Contains<RuntimeCall> for ContractsCallRuntimeFilter {
+    fn contains(call: &RuntimeCall) -> bool {
+        matches!(
+            call,
+            RuntimeCall::Staking(_) | RuntimeCall::NominationPools(_)
+        )
+    }
+}
+
 impl pallet_contracts::Config for Runtime {
     type Time = Timestamp;
     type Randomness = RandomnessCollectiveFlip;
     type Currency = Balances;
     type RuntimeEvent = RuntimeEvent;
     type RuntimeCall = RuntimeCall;
-    // The safest default is to allow no calls at all. This is unsafe experimental feature with no support in ink!
-    type CallFilter = Nothing;
+
+    type CallFilter = ContractsCallRuntimeFilter;
```

Comment:
What about the comment? Is it not true anymore? ie the feature now has support and is no longer experimental?

According to the [docs](https://github.com/Cardinal-Cryptography/polkadot-sdk/blob/master/substrate/frame/contracts/src/lib.rs#L269-L273) - it's now our job to never break the compatibility. This means we need some e2e test for that.

Reply:
"Safest is to allow no calls at all" -- this is still true and always will be :P 
"is unsafe experimental" -- this changed, it's stable

Yes, the compatibility thing is right, although it depends mainly on pallet staking and pallet nomination pools devs in substrate. If they make breaking changes we are screwed pretty much. We need to resolve such cases in a case-by-case manner. Having automated tests here is problematic:
- they are hard to write, and they will likely yield false positives
- they can only check API, and not the semantics. If semantics changes we are screwed anyway. 

What we need to do is monitor changes in these two pallets and upon every update need to make some specific checks. We have a similar procedure for releases -- I will raise this topic today on a release meeting, and try to add something to jira.

Reply:
Ok, I will try to add a test that will fail if the API changes. 

That being said, I don't think we should get a false feeling of security by having such a test. We need manual tests anyway.

Reply:
`matches!(call, X::Y(_))` will still not be enough - as long as the encoding produces something 'interpretable', we will not notice breaking change :disappointed:

Reply:
Ok, added tests. Please say they are perfect xD

Reply:
Beautiful.